presto-main/src/main/java/com/facebook/presto/split/SplitManager.java

```diff
@@ -54,20 +63,30 @@ public void removeConnectorSplitManager(ConnectorId connectorId)
         splitManagers.remove(connectorId);
     }
 
-    public SplitSource getSplits(Session session, TableLayoutHandle layout, SplitSchedulingStrategy splitSchedulingStrategy)
+    public SplitSource getSplits(Session session, TableHandle table, SplitSchedulingStrategy splitSchedulingStrategy)
     {
-        ConnectorId connectorId = layout.getConnectorId();
+        ConnectorId connectorId = table.getConnectorId();
         ConnectorSplitManager splitManager = getConnectorSplitManager(connectorId);
 
         ConnectorSession connectorSession = session.toConnectorSession(connectorId);
+        // TODO add support in metadata to check Connector still needs TableLayoutHandle and method in ConnectorSplitManager to use connectorTableHandle to get split
```

Comment:
nit: I don't quite understand this comment. Let's talk in person.

Reply:
```
// Now we will fetch the layout handle if it's not presented in TableHandle.
// In the future, ConnectorTableHandle will be used to fetch splits since it will contain layout information. 
```